### PR TITLE
fix(curriculum): hint and directions in RGP game step 37

### LINKED ISSUE #52677: Fix hint text and directions for RPG game step 37
### Describe the Issue

A few people on the [forum](https://forum.freecodecamp.org/search?q=step%2037%20RPG%20game) have been confused on how to handle step 37 of the RPG game. 

The common answer seems to be this

```js
let goStore = button1.onclick;
```
I think an easy fix is to update this current hint text and directions

## Proposed change

Change the third hint text to be the following:

```md
You should assign the `goStore` function reference to `button1.onclick`.
```

Also update the following portion of the directions to be the following to reference function instead of variable:

```md
Use dot notation to set the `onclick` property of your `button1` to the function reference of `goStore`. This function will be something you write later. Note that `button1` is already declared, so you do not need to use `let` or `const`.
```

## How to contribute

Since this is a small fix, I am opening this up for first timers only contribution.
Please read through the [contributing](https://contribute.freecodecamp.org/#/) docs first.
We do not assign issues. If you are interested in contributing, implement the change and create a PR.

Here is the file that you need to update

https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a3c2fccf186146b59c6e96.md

Only update the third hint text and none of the tests. Also update the last part of the directions as mentioned above. 

Happy coding! 

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-javascript-by-building-a-role-playing-game/step-37

### Your code

Campers are trying to pass with this code which is incorrect.
```
let goStore = button1.onclick;
```

### Expected behavior

We should update the hint text to my suggestion above to make it more clear. 

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sagarkashyap989**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #52677

- [x] i changed the third hint text 

- [x] i also changed the description 

this is my first open source contribution, please provide me some feedback to improve my work if it is needed

### CODE REVIEW

**Review by huyenltnguyen — COMMENTED** [review]
@sagarkashyap989 Thank you for the PR.

There is a change that I think we should make, but will need the Curriculum team to confirm, so let's hold off on that.

In the mean time, please help do the followings:
- Update the PR title to follow our convention: https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title
- Review the PR description and tick off the items that you have completed (I think you accidentally removed the checkboxes, and I added them back 🙂)

**Comment by sagarkashyap989** [comment]
> @sagarkashyap989 Thank you for the PR.
> 
> There is a change that I think we should make, but will need the Curriculum team to confirm, so let's hold off on that.
> 
> In the mean time, please help do the followings:
> 
> * Update the PR title to follow our convention: https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title
> * Review the PR description and tick off the items that you have completed (I think you accidentally removed the checkboxes, and I added them back 🙂)

- [x] updated PR title according to mentioned convention
- [x] ticked remaining checkboxes

thank you for you feedback

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a3c2fccf186146b59c6e96.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ You can access properties in JavaScript a couple of different ways. The first is
 button.onclick
 ```
 
-Use dot notation to set the `onclick` property of your `button1` to the variable `goStore`. This variable will be something you write later. Note that `button1` is already declared, so you do not need to use `let` or `const`.
+Use dot notation to set the `onclick` property of your `button1` to the function reference of `goStore`. This function will be something you write later. Note that `button1` is already declared, so you do not need to use `let` or `const`.
```
[the remark this answers is not in the document]
Comment (sagarkashyap989) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a3c2fccf186146b59c6e96.md:18]: > I've seen the Curriculum team updated the instructions to use contractions, so I guess we could have an improvement here as well.
> 
> Please don't make the change yet and wait for a confirmation from the team.

Thank you for pointing that out! I'll hold off on making the change for now and will wait for confirmation from the team before proceeding. Let me know if there's anything else I should consider. Appreciate your guidance

[the remark this answers is not in the document]
Comment (Sembauke) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a3c2fccf186146b59c6e96.md:18]: Yep this change is fine, you can go ahead and add it.

**Review by huyenltnguyen — APPROVED** [review]

**Review by gikf — APPROVED** [review]

**Comment by gikf** [comment]
Hi @sagarkashyap989. Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Pull request merged**